File: spsc-lite-ruby/ProcessTree_Tests.rb
require 'test/unit'
require 'SLL'
require 'ProcessTree'

class ProcessTree_Tests < Test::Unit::TestCase
  include SLL
  include ProcessTree
  def setup()
    t = Tree.new(Var.new("r"))
    r = t.root
    t.addChildren(r,[[Var.new("m1"), nil], [Var.new("m2"), nil]])
    m1 = r.children[0]
    m2 = r.children[1]
    t.addChildren(m1, [[Var.new("n"), nil]])
    t.replaceSubtree(m2, Var.new("x"))
    @tree = t
  end

  def test01PrTreeBuilding()
    assert_equal(
    "{0:(r,,,[1,2]),1:(m1,,0,[3]),3:(n,,1,[]),2:(x,,0,[])}",
    @tree.to_s)
  end

  def test02PrTreeNodes()
    assert_equal([0,1,3,2], @tree.nodes.map{|n| n.nodeId})
  end

  def test03PrTreeLeaves()
    assert_equal([3,2], @tree.leaves.map{|n| n.nodeId})
  end

end

File: spsc-lite-python/src/spsc/process_tree_tests.py
'''
Created on Aug 18, 2009

@author: Sergei Romanenko
'''

import unittest

from sll_language import *
from process_tree import *

class ProcessTreeTest(unittest.TestCase):

    def setUp(self):
        t = ProcessTree(Var("r"))
        r = t.root
        t.addChildren(r,[(Var("m1"), None), (Var("m2"), None)])
        m1 = r.children[0]
        m2 = r.children[1]
        t.addChildren(m1, [(Var("n"), None)])
        t.replaceSubtree(m2, Var("x"))
        self.tree = t

    def test01PrTreeBuilding(self):
        "PrTreeBuilding"
        self.assertEqual("{0:(r,None,None,[1,2]),1:(m1,None,0,[3]),3:(n,None,1,[]),2:(x,None,0,[])}",
                         "%s" % self.tree)

    def test02PrTreeNodes(self):
        "PrTreeNodes"
        self.assertEqual([0,1,3,2], [n.nodeId for n in self.tree.nodes()])

    def test03PrTreeLeaves(self):
        "PrTreeLeaves"
        self.assertEqual([3,2], [n.nodeId for n in self.tree.leaves()])
